types/three/src/renderers/webgl/WebGLPrograms.d.ts

```diff
@@ -20,9 +25,16 @@ export class WebGLPrograms {
 
     programs: WebGLProgram[];
 
-    getParameters(material: Material, lights: any, shadows: object[], scene: Scene, object: any): any;
-    getProgramCacheKey(parameters: any): string;
-    getUniforms(material: Material): object;
-    acquireProgram(parameters: any, cacheKey: string): WebGLProgram;
+    getParameters(
+        material: Material,
+        lights: WebGLLightsState,
+        shadows: Light[],
+        scene: Scene,
+        object: Object3D,
+    ): WebGLProgramParameters;
```

Comment:
- `lights` comes from [here](https://github.com/mrdoob/three.js/blob/4acb78002425969f7c0cf46423ea5f728c41dfbc/src/renderers/WebGLRenderer.js#L1598) (the `state` property is passed into `getParameters`)
- `shadows` comes from [here](https://github.com/mrdoob/three.js/blob/4acb78002425969f7c0cf46423ea5f728c41dfbc/src/renderers/WebGLRenderer.js#L1599) and is populated with Lights [here](https://github.com/mrdoob/three.js/blob/4acb78002425969f7c0cf46423ea5f728c41dfbc/src/renderers/WebGLRenderer.js#L965) and [here](https://github.com/mrdoob/three.js/blob/4acb78002425969f7c0cf46423ea5f728c41dfbc/src/renderers/WebGLRenderer.js#L983).
- `object` comes from [here](https://github.com/mrdoob/three.js/blob/4acb78002425969f7c0cf46423ea5f728c41dfbc/src/renderers/WebGLRenderer.js#L999-L1025) and should be `Object3D`.